File: greedy_change.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    //Get dollar input
    float dollars;
    do
    {
        dollars = get_float("Change owed?: ");
    }
    while (dollars < 0);
    int cents = round(dollars * 100);

    //Initialize variable counting coins for change
    int minCoins = 0;

    //Change using minimum number of coins
    while (cents > 0)
    {
        //Use quarters first
        if (cents >= 25)
        {
            cents -= 25;
            minCoins += 1;
        }
        //Then dimes
        else if (cents >= 10)
        {
            cents -= 10;
            minCoins += 1;
        }
        //Then nickels
        else if (cents >= 5)
        {
            cents -= 5;
            minCoins += 1;
        }
        //Then pennies
        else if (cents >= 1)
        {
            cents -= 1;
            minCoins += 1;
        }
    }

    printf("%i\n", minCoins);
}

File: CS50/greedy_change.py
from cs50 import get_float

while True:
    dollars = get_float("Change owed: ")
    if dollars > 0:
        break
change = round(dollars * 100)

# Variable to track number of coins
minCoins = 0

while change > 0:
    # Quarters first
    if change >= 25:
        change -= 25
        minCoins += 1

    # Then dimes
    elif change >= 10:
        change -= 10
        minCoins += 1

    # Then nickels
    elif change >= 5:
        change -= 5
        minCoins += 1

    # Then pennies
    elif change >= 1:
        change -= 1
        minCoins += 1

print(minCoins)
